internal/clusterconfig/clusterconfig.go

```diff
@@ -145,6 +160,17 @@ func (r *UpgradeClusterConfigGather) writeSecretToFile(secret *corev1.Secret, fi
 	return utils.WriteToFile(secret, filePath)
 }
 
+func (r *UpgradeClusterConfigGather) writeProxyToFile(proxy *v1.Proxy, filePath string) error {
+	r.Log.Info("Writing proxy to file", "path", filePath)
+	p := &v1.Proxy{
```

Comment:
why not to remove metadata from original proxy with typeMetaForObject?

Reply:
I didn't go with the `typeMetaForObject` method, because I wanted to leave out the [ObjectMeta](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L111) of the original object, while IIUC the former replaces the object's [TypeMeta](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L42). WDYT?